src/float.jl

```diff
@@ -0,0 +1,352 @@
+    # The unbiased exponent is <= 0
+    if exponent <= bias(T)
+        # reverse the bits of the mantissa in place
+        bitreverse(mantissa) >> (exposize(T) + 1)
+    elseif exponent >= fracsize(T) + bias(T)
+        mantissa
+    else
+        # reverse the low bits of the fractional part
+        # as determined by the exponent
+        n_reverse_bits = fracsize(T) + bias(T) - exponent
+        # isolate the bits to be reversed
+        to_reverse = mantissa & iT((1 << n_reverse_bits) - 1)
+        # zero them out
+        mantissa = mantissa ⊻ to_reverse
+        # reverse them and put them back in place
+        mantissa |= bitreverse(to_reverse) >> (8 * sizeof(T) - n_reverse_bits)
+    end
+end
+
+
+"""
+    lex_to_float(T, bits)
+
+Reinterpret the bits of a floating point number using an encoding with better shrinking
+properties.
+This produces a non-negative floating point number, possibly including `NaN` or `Inf`.
+
+The encoding is ported from Hypothesis, and has the property that lexicographically smaller
+bit patterns correspond to 'simpler' floats.
+
+# Encoding
+
+The encoding used is as follows:
+
+If the sign bit is set: 
+
+    - the remainder of the first byte is ignored
+    - the remaining bytes are interpreted as an integer and converted to a float
+
+If the sign bit is not set:
+
+    - the exponent is encoded using [`encoded_exponent`](@ref)
+    - the mantissa is updated using [`update_mantissa`](@ref)
+    - the float is reassembled using [`assemble`](@ref)
+
+## Extended Help
+
+See the [reference implementation](https://github.com/HypothesisWorks/hypothesis/blob/aad70fb2d9dec2cef9719cdf5369eec9fae0d2a4/hypothesis-python/src/hypothesis/internal/conjecture/floats.py#L176) in Hypothesis.
+
+"""
+function lex_to_float(::Type{T}, bits::I)::T where {I,T<:Base.IEEEFloat}
+    sizeof(T) == sizeof(I) || throw(ArgumentError("The bitwidth of `$T` needs to match the bitwidth of the given bits!"))
+    iT = uint(T)
+    sign, exponent, mantissa = tear(reinterpret(T, bits))
+    if isone(sign)
+        exponent = encode_exponent(exponent)
+        mantissa = update_mantissa(T, exponent, mantissa)
+        assemble(T, zero(iT), exponent, mantissa)
+    else
+        integral_mask = signed(iT)(-1) >>> 0x8
+        integral_part = bits & integral_mask
+        T(integral_part)
+    end
+end
+
+"""
+    float_to_lex(f)
+
+Encoding a floating point number as a bit pattern.
+
+This is essentially the inverse of [`lex_to_float`](@ref) and produces a bit pattern that is lexicographically
+smaller for 'simpler' floats.
+
+Note that while `lex_to_float` can produce any valid positive floating point number, it is not injective. So combined with the fact that positive and negative floats map to the same bit pattern,
+`float_to_lex` is not an exact inverse of `lex_to_float`.
+"""
+function float_to_lex(f::T) where {T<:Base.IEEEFloat}
+    # If the float is simple, we can just reinterpret it as an integer
```

Comment:
```suggestion
    # If the float is simple, we can just convert it to an integer
```

`reinterpret` of a float maps to quite different numbers.